### PR TITLE
[Fleet] Fix agent enrollment script to not create a new enrollment token

### PR DESCRIPTION
**Pull request by nchaulet**
## Description

This script is creating a new enrollment token associated to a non existing config (`default` is not a valid id, the id of the default config is a random uuid)

I updated this script to get the default enrollment token created during fleet setup instead.

Related to https://github.com/elastic/kibana/issues/65669

Also created an issue on kibana  https://github.com/elastic/kibana/issues/65752 to not allow the creation of an enrollment token for a non existing agent config id.

### CODE REVIEW

File: dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl [hunk 1]
```diff
@@ -30,19 +30,31 @@ function enroll(){
     if [[ -n "${FLEET_ENROLLMENT_TOKEN}" ]] && [[ ${FLEET_ENROLLMENT_TOKEN} == 1 ]]; then
       apikey = "${FLEET_ENROLLMENT_TOKEN}"
     else
-      enrollResp=$(curl -X POST  ${KIBANA_HOST:-http://localhost:5601}/api/ingest_manager/fleet/enrollment-api-keys \
+      enrollResp=$(curl -X GET  ${KIBANA_HOST:-http://localhost:5601}/api/ingest_manager/fleet/enrollment-api-keys \
```
Comment (nchaulet) [inline dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl:33]: this response return the list of API keys but not the secret API key that you can only get doing a specific request to the ressource

Comment (mtojek) [inline dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl:33]: Is the "-X GET" a default option? I think we can skip it (in all places).

**Comment by ruflin** [comment]
Is this a bug that should also be backported to 7.8?

**Comment by nchaulet** [comment]
Yes this should probably backported 👍

**Comment by michalpristas** [comment]
_testing this today_

**Comment by nchaulet** [comment]
@michalpristas I am not sure why test are failing here, can I get some help?

**Comment by michalpristas** [comment]
@nchaulet probably flakyness i restarted test, try merging with master

**Comment by nchaulet** [comment]
@elasticmachine merge upstream

**Comment by michalpristas** [comment]
failures do not seem related, locally libbeat passes, i think it's ok to get this in

**Pull request merged**